### ISSUE TITLE
Fix generating the `.idea` folder for Intellij users of the `scala-cli` repo

### ISSUE THREAD

**Issue by Gedochao** [VirtusLab/scala-cli#936]
**Version(s)**

N/A

**Describe the bug**
Running `./mill mill.scalalib.GenIdea/idea` no longer works for main, instead returning the following output.
It seems that there's an issue with fetching `scalafmt-cli_3:3.5.1`.

```bash
~/IdeaProjects/scala-cli  main ✗                                                                                                                                                                  3d2h ⚑ ◒  ⍉
▶ ./mill mill.scalalib.GenIdea/idea
[1/1] mill.scalalib.GenIdea.idea 
Analyzing modules ...
[1/1] mill.scalalib.GenIdea.idea > [164/2388] de.tobiasroeser.mill.vcs.version.VcsVersion.vcsState.overridden.de.tobiasroeser.mill.vcs.version.VcsVersion.vcsState 
[1/1] mill.scalalib.GenIdea.idea > [2192/2388] dummy.scalafmt.resolvedIvyDeps | Downloading [2/2] artifacts (~0/0 bytes)
1 targets failed
mill.scalalib.GenIdea.idea Failure during resolving modules: dummy.scalafmt.resolvedIvyDeps 
Resolution failed for 1 modules:
--------------------------------------------
  org.scalameta:scalafmt-cli_3:3.5.1 
        not found: /Users/pchabelski/.ivy2/local/org.scalameta/scalafmt-cli_3/3.5.1/ivys/ivy.xml
        not found: https://repo1.maven.org/maven2/org/scalameta/scalafmt-cli_3/3.5.1/scalafmt-cli_3-3.5.1.pom

```

**To Reproduce**

Just run
```bash
./mill mill.scalalib.GenIdea/idea
```
**Expected behaviour**
`mill` should generate the correct `.idea` folder for the `scala-cli` project

**Comment by Gedochao** [comment]
Seems to have been fixed already, since I no longer can replicate it on `main` branch.
It might still be worth it to do #937 to prevent similar issues in the future.
closing this.

**Comment by mtk** [comment]
hmm.... in a brand new user ID with no cached state, downloaded coursier, did a 'setup', cloned VirtusLab/scala-cli (29e0b3a33dc9b432c85a57a226bfdce03d15992a)
 and did a './mill mill.scalalib.GenIdea/idea' and ran into this problem.  is this related or new?  it is repeatable.  or should i submit a new issue?

<details>
<summary>Details</summary>

1 targets failed
mill.scalalib.GenIdea.idea Failure during resolving modules: integration.modulesPath java.nio.file.FileSystemException: /home/testy/scala-cli/out/mill-worker-lsXbgnYzGxiOjE19ueHbpE/+PLU=-1/mill-2f64853d03589440a01e0be21259a457-io: No such device or address
    java.base/sun.nio.fs.UnixException.translateToIOException(UnixException.java:100)
    java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:106)
    java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:111)
    java.base/sun.nio.fs.UnixFileSystemProvider.newByteChannel(UnixFileSystemProvider.java:219)
    java.base/java.nio.file.Files.newByteChannel(Files.java:380)
    java.base/java.nio.file.Files.newByteChannel(Files.java:432)
    java.base/java.nio.file.spi.FileSystemProvider.newInputStream(FileSystemProvider.java:422)
    java.base/java.nio.file.Files.newInputStream(Files.java:160)
    os.Path.getInputStream(Path.scala:491)
    os.read$inputStream$.apply(ReadWriteOps.scala:229)
    mill.api.PathRef$.$anonfun$apply$2(PathRef.scala:48)
    mill.api.PathRef$.$anonfun$apply$2$adapted(PathRef.scala:38)
    scala.collection.IterableOnceOps.foreach(IterableOnce.scala:563)
    scala.collection.IterableOnceOps.foreach$(IterableOnce.scala:561)
    scala.collection.AbstractIterable.foreach(Iterable.scala:926)
    scala.collection.IterableOps$WithFilter.foreach(Iterable.scala:896)
    mill.api.PathRef$.apply(PathRef.scala:38)
    ammonite.$file.build$CliIntegration.$anonfun$modulesPath$2(build.sc:680)
    mill.define.Task$TraverseCtx.evaluate(Task.scala:380)
(END)

</details>

**Comment by lwronski** [comment]
Hi @mtk, thanks for reporting 

Did you try cleaning the project directory before running it? so, you should remove `out` in `scala-cli` directory. 

If that doesn't help, please create a new issue because it is not related problem to the current issue.

**Comment by mtk** [comment]
just got back to this and git bisect'ed it.  it was the update from mill 0.10.3 to 0.10.4 that broke GenIdea/idea and Bloop/install.  i will create a separate issue.

**Comment by lwronski** [comment]
> just got back to this and git bisect'ed it. it was the update from mill 0.10.3 to 0.10.4 that broke GenIdea/idea and Bloop/install. i will create a separate issue.

It should be fixed by #1012

**Comment by mtk** [comment]
thanks!  wasn't 0.10.3->0.10.4 at all (i misread something in gitk).  i
also reported this as a mill issue and the misleading error message was
verified.  they're working on a fix.
